### PR TITLE
Migrate GSN to solidity 0.6.2

### PR DESCRIPTION
**Pull request by forshtat**
Note: 0x LibBytes was checked-in here

### CODE REVIEW

File: contracts/interfaces/IRelayRecipient.sol [hunk 1]
```diff
@@ -1,17 +1,17 @@
-pragma solidity ^0.5.16;
+pragma solidity ^0.6.2;
 
 /**
  * a contract must implement this interface in order to support relayed transaction.
  * It is better to inherit the BaseRelayRecipient as its implementation.
  */
-contract IRelayRecipient {
+abstract contract IRelayRecipient {
```
Comment (shahafn) [inline contracts/interfaces/IRelayRecipient.sol:7]: why not interface?

File: package.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
   },
   "prepublish": "npm run prepare",
   "scripts": {
-    "test": "yarn run lint && yarn extract_abi && yarn generate && yarn tsc && npm run test-js",
+    "test": "yarn run lint && yarn run truffle-compile && yarn extract_abi && yarn generate && yarn tsc && npm run test-js",
```
Comment (shahafn) [inline package.json:33]: 1. `yarn generate` already compiles.
2. If you're at it, then it's weird that we have both `yarn run` and `npm run`, even in the same line.

File: contracts/interfaces/IRelayRecipient.sol (hunk 1 above)
Comment (forshtat) [reply contracts/interfaces/IRelayRecipient.sol:7]: A) it was a contract
B) I think it has to call this._msgSender() for interface

Comment (drortirosh) [reply contracts/interfaces/IRelayRecipient.sol:7]: Interface can't declare"internal" methods

Comment (drortirosh) [reply contracts/interfaces/IRelayRecipient.sol:7]: Interface can't declare"internal" methods

Comment (drortirosh) [reply contracts/interfaces/IRelayRecipient.sol:7]: Interface can't declare"internal" methods

**Pull request merged**